packages/instant/src/util/gas_price_estimator.ts

```diff
@@ -0,0 +1,44 @@
+import { BigNumber } from '@0x/utils';
+
+import { DEFAULT_GAS_PRICE } from '../constants';
+
+interface GasStationResult {
+    average: number;
+    fastestWait: number;
+    fastWait: number;
+    fast: number;
+    safeLowWait: number;
+    blockNum: number;
+    avgWait: number;
+    block_time: number;
+    speed: number;
+    fastest: number;
+    safeLow: number;
+}
+
+const endpointUrl = 'https://ethgasstation.info/json/ethgasAPI.json';
+const fetchFastAmountInWei = async () => {
+    const res = await fetch(endpointUrl);
+    const gasInfo = (await res.json()) as GasStationResult;
+    const gasPriceInGwei = new BigNumber(gasInfo.fast / 10);
+    return gasPriceInGwei.mul(1000000000);
+};
+
+export class GasPriceEstimator {
+    private _lastFetched?: BigNumber;
+    public async getFastAmountInWeiAsync(): Promise<BigNumber> {
+        let fetchedAmount: BigNumber | undefined;
+        try {
+            fetchedAmount = await fetchFastAmountInWei();
+        } catch {
+            fetchedAmount = undefined;
+        }
+
+        if (fetchedAmount) {
+            this._lastFetched = fetchedAmount;
+        }
+
+        return fetchedAmount || this._lastFetched || DEFAULT_GAS_PRICE;
```

Comment:
If I'm reading this correctly, the cached value is just for cases when the latest request fails? If so we probably don't need the warmup step

Reply:
Yes.  I like the warmup step though, because if the fetch fails on our first time submitting a txn, we have a value to fall back to instead of having to rely on the default.

Reply:
Thinking about this a bit more, I'm wondering if the warmup is worth it, esp. if this gets a lot of txn and starts hammering ethgasstation